src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11114,7 +11071,43 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                 }
 
                 // We'll "take" this operator, as precedence is tentatively OK.
-                var opToken = this.EatContextualToken(tk);
+
+                SyntaxToken opToken;
+
+                // Combine tokens into a single token if needed
+                if (tokensToCombine == 1)
```

Comment:
reordered so that we don't unilaterally eat a token befure doing the merging.  It allows us to more cleanly use helpers like EatDotDotToken which want to be on the initial dot.